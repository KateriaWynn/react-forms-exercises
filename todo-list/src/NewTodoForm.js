import React, { useState } from 'react';

/** Form for creating a new todo to add to a todo list.
 *
 * Has state for the task for each todo on submission,
 * sends { todo } to function (addTodo) received from parent (TodoList).
 *
 */

function NewTodoForm({ addTodo }) {
  const INITIAL_STATE = { task: 'Complete React assignments' };
  const [formData, setFormData] = useState(INITIAL_STATE);

  /** Send { task } to parent & clear form. */
  const handleSubmit = (evt) => {
    evt.preventDefault();
    addTodo(formData);
    setFormData(INITIAL_STATE);
  };

  /** Update local state w/curr state of input element */

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  /** render form */

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="task">New todo: </label>
      <input
        type="text"
        id="task"
        name="task"
        value={formData.task}
        onChange={handleChange}
      />

      <button>Add new todo</button>
    </form>
  );
}

export default NewTodoForm;
