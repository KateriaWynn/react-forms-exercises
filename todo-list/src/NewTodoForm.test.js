import React from 'react';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// smoke test
test('if NewTodoForm renders without crashing', () => {
  render(<NewTodoForm />);
});

//snapshot test
test('if NewTodoForm matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
