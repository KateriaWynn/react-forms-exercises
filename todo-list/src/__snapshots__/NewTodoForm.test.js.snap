// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if NewTodoForm matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      New todo:
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value="Complete React assignments"
    />
    <button>
      Add new todo
    </button>
  </form>
</DocumentFragment>
`;
